### ISSUE TITLE
card on file processing for non-recurring payments

### ISSUE THREAD

**Issue by lcdservices** [iATSPayments/com.iatspayments.civicrm#205]
the card on file functionality (https://github.com/iATSPayments/com.iatspayments.civicrm/wiki/Card-on-File) lets you process a one time contribution as part of a recurring contribution cycle -- to "catch up" donors, etc.

one of my clients is interested in that type of functionality for one-time contributions that are not tied to a recurring payment. for some of their events (which are actually multi-day trips) they ask registrants to commit to the full cost of the trip (the registrants then raise money via PCPs to help cover the cost). they want the ability to store the person's credit card as part of that initial commitment, and then run the card at a later time once that actual balance owed is determined (a factor of how much fundraising they do).

it would seem that a framework is already in place to do this, but it would need to be adapted for this use case. is there any interest in providing this type of feature in the extension? we may be able to help sponsor the development.

**Comment by KarinG** [comment]
Similar - though slightly different and with the tools currently available - to get people's cards authorized ahead of time to enure a faster checkout an a fundraising event:

- before the event: online registration form pre-configured for: $1 contribution - recurring - installments: 2; frequency annual
- after the event: imported amounts and next_scheduled_contribution dates to update the recurring series
- they all transacted next day - and series were completed

**Comment by adixon** [comment]
Yeah, I had a customer with an almost identical requirement, and you're right that the infrastructure is almost all there.

I think the easiest route would be to create a recurring contribution with only one contribution, which you sort of already have with the new future recurring public front end option.

If you want to try that out from a user experience and think about how it might be tinkered with to cover your use case, I'd be happy to implement it.
